auth/protocol.go

```diff
@@ -60,6 +60,60 @@ type PublicKeyAuthRequest struct {
 	PublicKey string `json:"publicKey"`
 }
 
+// AuthorizationRequest is the authorization request used after some
+// authentication methods (e.g. kerberos) to determine whether users are
+// allowed to access the service
+//
+// swagger:model AuthorizationRequest
+type AuthorizationRequest struct {
+	// PrincipalUsername is the authenticated username of the user, this
+	// username has been verified to be correct and to correspond to the
+	// user that is connecting.
+	//
+	// required: true
+	PrincipalUsername string `json:"principalUsername"`
+	// LoginUsername is the username the user wishes to log in as. In
+	// general, the authorization must check that this matches the
+	// PrincipalUsername, however in some cases it may be beneficial to let
+	// some users log in as others (e.g. administrators logging in as
+	// normal users to debug)
+	//
+	// required: true
+	LoginUsername string `json:"loginUsername"`
+	// RemoteAddress is the address the user is connecting from
+	//
+	// required: true
+	RemoteAddress string `json:"remoteAddress"`
+	// ConnectionID is an opaque ID to identify the SSH connection
+	//
+	// required: true
+	ConnectionID string `json:"connectionId"`
+}
+
+const  (
+	// METADATA_INTERAL signifies that a value is internal and for use only in containerssh
+	METADATA_INTERNAL     = 0
+	// METADATA_CONFIG signifies that a value can be exposed to the configuration server
+	METADATA_CONFIG       = 1 << 0
+	// METADATA_CONTAINER signifies that a value can be exposed to the pod as environment variables or files
+	METADATA_CONTAINER    = 1 << 1
+)
+
+type ConnectionMetadata struct {
+	// Metadata is a set of key-value pairs that can be returned and
+	// either consumed by the configuration server or exposed in the
+	// backend as environment variables.
+	Metadata    map[string]string `json:"metadata,omitempty"`
+	// Files is a key-value pair of files to be placed inside containers.
+	// The key represents the path to the file while the value is the
+	// binary content.
+	Files       map[string][]byte `json:"files,omitempty"`
+	// Sensitivity is a set of key-value pairs that indicates the
+	// sensitivity of a value. The key identifies the entry name and is
+	// valid for all maps in ConnectionMetadata
+	Sensitivity map[string]int    `json:"sensitivity,omitempty"`
```

Comment:
I ended up removing sensitivity entirely. Since the credentials are a file now only `Metadata` is send to the webhooks, environment and files stay only in the server and are send to the container at the end of the chain.